press/utils/billing.py

```diff
@@ -126,16 +126,35 @@ def get_publishable_key():
 
 
 def get_setup_intent(team):
+	from frappe.utils import random_string
+
 	intent = frappe.cache().hget("setup_intent", team)
 	if not intent:
-		customer_id = frappe.db.get_value("Team", team, "stripe_customer_id")
+		data = frappe.db.get_value("Team", team, ["stripe_customer_id", "currency"])
+		customer_id = data[0]
+		currency = data[1]
 		stripe = get_stripe()
+		hash = random_string(10)
 		intent = stripe.SetupIntent.create(
 			customer=customer_id,
 			payment_method_types=["card"],
-			payment_method_options={"card": {"request_three_d_secure": "automatic"}},
+			payment_method_options={
+				"card": {
+					"request_three_d_secure": "automatic",
+					"mandate_options": {
+						"reference": f"Mandate-team:{team}-{hash}",
```

Comment:
reference name can be better. I don't like this much. Need suggestion.

Reply:
according to docs, `reference` should be unique string to identity this mandate. so random hash is not useful.
we could just use team_id with amount?

Reply:
since the mandate is being created on `SetupIntent` and not during invoice creation we cannot use amount as it is not determined and the maximum amount limit would be same for all the teams.